src/loophole.coffee

```diff
@@ -2,19 +2,35 @@ vm = require 'vm'
 
 exports.allowUnsafeEval = (fn) ->
   previousEval = global.eval
+  callback = -> global.eval = previousEval
+  error = false
   try
     global.eval = (source) -> vm.runInThisContext(source)
-    fn()
+    fn(callback)
```

Comment:
Sorry, I was mistaken with this comment. Still, is there any way to simplify the control flow around the exception handling?